insights/v1alpha1/types_insights.go

```diff
@@ -235,19 +299,23 @@ type ObjectReference struct {
 	// This value should consist of only lowercase alphanumeric characters, hyphens and periods.
 	// Example: "", "apps", "build.openshift.io", etc.
 	// +kubebuilder:validation:Pattern:="^$|^[a-z0-9]([-a-z0-9]*[a-z0-9])?(.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$"
+	// +kubebuilder:validation:MaxLength=512
 	// +required
 	Group string `json:"group"`
 	// resource is the type that is being referenced.
 	// It is normally the plural form of the resource kind in lowercase.
 	// This value should consist of only lowercase alphanumeric characters and hyphens.
 	// Example: "deployments", "deploymentconfigs", "pods", etc.
-	// +required
 	// +kubebuilder:validation:Pattern:="^[a-z0-9]([-a-z0-9]*[a-z0-9])?$"
+	// +kubebuilder:validation:MaxLength=512
+	// +required
 	Resource string `json:"resource"`
 	// name of the referent.
+	// +kubebuilder:validation:MaxLength=256
 	// +required
 	Name string `json:"name"`
 	// namespace of the referent.
+	// +kubebuilder:validation:MaxLength=512
```

Comment:
Why 512?

Reply:
Changed to 253 base on k8s naming convention